lib/PSR/Events/Interface.hs

```diff
@@ -50,11 +56,40 @@ data EventFilterParams = EventFilterParams
     }
     deriving (Generic)
 
-data Events = Events 
+data Events = Events
     { addExecutionEvent :: BlockHeader -> ExecutionEventPayload -> IO ()
     , addCancellationEvent :: BlockHeader -> ScriptHash -> IO ()
     , addSelectionEvent :: BlockHeader -> IO ()
-    , getEventsChannel :: TChan Event 
+    , getEventsChannel :: TChan Event
     , getEvents :: EventFilterParams -> IO [Event]
     }
 
+eventMatchesFilter :: EventFilterParams -> Event -> Bool
+eventMatchesFilter (EventFilterParams typ time_begin time_end slot_begin slot_end _limit _offset name_or_script_hash) event =
```

Comment:
May be better to rename `_eventFilterParam_` prefix with something shorter.
Since most of the types of the record fields are the same, it's very easy to mess up the order.